pkg/gather/clusterconfig/0_gatherer.go

```diff
@@ -160,6 +142,38 @@ func (g *Gatherer) Gather(ctx context.Context, gatherList []string, recorder rec
 	return nil
 }
 
+func (g *Gatherer) gatherStatusReport(gather string, recorder recorder.Interface, starts time.Time, rv reflect.Value) (gatherStatusReport, []string) {
+	var errorsReport []string
+	elapsed := time.Since(starts).Truncate(time.Millisecond)
+
+	gatherResults, _ := rv.Interface().(gatherResult)
+	gatherFunc := gatherFunctions[gather].function
+	gatherCanFail := gatherFunctions[gather].canFail
+	gatherName := runtime.FuncForPC(reflect.ValueOf(gatherFunc).Pointer()).Name()
+
+	klog.V(4).Infof("Gather %s took %s to process %d records", gatherName, elapsed, len(gatherResults.records))
+
+	statusReport := gatherStatusReport{gatherName, time.Duration(elapsed.Milliseconds()), len(gatherResults.records), extractErrors(gatherResults.errors)}
+
+	if gatherCanFail {
+		for _, err := range gatherResults.errors {
+			klog.V(5).Infof("Couldn't gather %s' received following error: %s\n", gatherName, err.Error())
+		}
+	} else {
+		errorsReport = extractErrors(gatherResults.errors)
+	}
+
+	for _, record := range gatherResults.records {
```

Comment:
wouldn't it makes sense to have the "recording" part in a separate function or just rename the function.
because the name `gatherStatusReport` is a bit misleading, this functions can be better described as `handleResult` or something. Also it doesn't uses the `g` receiver anywhere so I just wouldn't make it a method.

Reply:
I think that splitting in an even smaller function could be too much, but I agree with the rename. How about renaming it to `handleStatusReport`?

Reply:
It creates the `statusReport` and stores the `gatherResult` neither of which is described by  `handleStatusReport` :sweat_smile: 
I would go with `handleGatherResult` and have the `reflect.Value` be already be converted to `gatherResult` before calling it.
Something like: 
```
func handleGatherResult(gather string, gr gatherResult, recorder recorder.Interface, starts time.Time) (gatherStatusReport, []string) {

```

Reply:
I mean, fine with the name change, and it does make sense, but as you told, it creates the report. Probably would be better to call it `createGatherReport` instead. 

If you check the function response is that what it is returning. I'm not sure about passing the `gatherResult`, otherwise, you will have to put some other parts back to that other function not actually reducing the code complexity.

For example, the `gatherCanFail` variable is something that you will have to pass as a parameter as well, then you will have 5 parameters in the function call, which is way too much (I mean, IMHO 4 is already more than it should have. Code clean says that <= 3 is fine.).

But I can see where are you going. Maybe there is some space to improve it, let me try something.

Reply:
@0sewa0 hopefully, I may have improved it.

Reply:
much improved, thx :smile: